Bottom: d4e6a547957aa257a74e0a3531c5a68f364a5645
Top:    8a4d710165bd56da8fcf1832c41bb484e2c64b94
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-12 17:36:16 +0100

GTK HID: Don't use GObject's "constructed" member

This member, although convenient, was added in GLib 2.12.
PCB only checks for GLib >2.4.


---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 2f80be7..53f2a81 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -255,23 +255,36 @@ enum
 static GObjectClass *ghid_pinout_preview_parent_class = NULL;
 
 
-/*! \brief GObject constructed handler
+/*! \brief GObject constructor
  *
  *  \par Function Description
- *  Setup any the necessary state for our widget now it is constructed.
+ *  Chain up and construct the object, then setup the
+ *  necessary state for our widget now it is constructed.
  *
- *  \param [in] widget  The GObject having just been constructed.
+ *  \param [in] type                    The GType of object to be constructed
+ *  \param [in] n_construct_properties  Number of construct properties
+ *  \param [in] contruct_params         The construct properties
+ *
+ *  \returns The GObject having just been constructed.
  */
-static void
-ghid_pinout_preview_constructed (GObject *object)
+static GObject *
+ghid_pinout_preview_constructor (GType type,
+                                 guint n_construct_properties,
+                                 GObjectConstructParam *construct_properties)
 {
-  GtkWidget *widget = GTK_WIDGET (object);
+  GObject *object;
 
-  gtk_widget_set_gl_capability (widget,
+  /* chain up to constructor of parent class */
+  object = G_OBJECT_CLASS (ghid_pinout_preview_parent_class)->
+    constructor (type, n_construct_properties, construct_properties);
+
+  gtk_widget_set_gl_capability (GTK_WIDGET (object),
                                 gport->glconfig,
                                 NULL,
                                 TRUE,
                                 GDK_GL_RGBA_TYPE);
+
+  return object;
 }
 
 
@@ -367,7 +380,7 @@ ghid_pinout_preview_class_init (GhidPinoutPreviewClass * klass)
   GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
   GtkWidgetClass *gtk_widget_class = GTK_WIDGET_CLASS (klass);
 
-  gobject_class->constructed = ghid_pinout_preview_constructed;
+  gobject_class->constructor = ghid_pinout_preview_constructor;
   gobject_class->finalize = ghid_pinout_preview_finalize;
   gobject_class->set_property = ghid_pinout_preview_set_property;
   gobject_class->get_property = ghid_pinout_preview_get_property;
